src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultTagHelperComponentContextInitializer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Razor.TagHelpers;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    /// <summary>
+    /// Default implementation of <see cref="ITagHelperComponentContextInitializer"/>.
+    /// </summary>
+    public class DefaultTagHelperComponentContextInitializer : ITagHelperComponentContextInitializer
+    {
+        /// <inheritdoc />
+        public ITagHelperComponent InitializeViewContext(ITagHelperComponent tagHelperComponent, ViewContext context)
```

Comment:
Continuing off my comment on the class I'd probably also change the name of this method. Also, since it's just initializing i wouldn't return `ITagHelperComponent` because that makes it feel like the API creates a new one. Suggest `void Initialize(ITagHelperComponent, ViewContext)`. This is more in line with other public apis we have about creating/activating/initializing a specific type.